alg/gdalgrid.cpp

```diff
@@ -1901,6 +2149,9 @@ GDALGridContextCreate( GDALGridAlgorithm eAlgorithm, const void *poOptions,
             memcpy(poOptionsNew, poOptions, sizeof(GDALGridDataMetricsOptions));
 
             pfnGDALGridMethod = GDALGridDataMetricMinimum;
+            bCreateQuadTree = (nPoints > 100 &&
```

Comment:
I believe we should also add dfAngle == 0 as a condition here and in the below case (including for the nearest neighbour case), as the formulas in the quadtree case don't take into account rotation.

Reply:
@rouault Yes, of course, thsi would make absolute sense.